roles/adr/meta/main.yml

```diff
@@ -12,9 +12,3 @@ galaxy_info:
       versions:
         - 20.04
   galaxy_tags: []
-
-# Although technically, adr needs neither base or java to install, JAVA_HOME needs to be set in order for the app to
-# be able to run.
-dependencies:
-  - role: base
-  - role: java
```

Comment:
I don't mind removing base from packages, as long as it's not needed - but removing roles that are actually required for this to work seems unfortunate. The alternative is either documentation (which no one reads) or pre-install checks/conditional installs which seem verbose.

Reply:
I understand the pick/chose idea, but deploying adr without deploying _a_ java instance is pointless. Of course, the user can have deployed java on their own (very likely), but then we basically have to check for the presence of java.

Similar issue with arcanist. You can install the dependencies "manually" (and things like git almost certainly are), but in the case where they aren't, the user is going to get frustrating errors, and it won't be clear why they're getting those errors. I'd rather not have to waste time supporting those kind of issues.

Reply:
I tend to agree with this change actually.
We've discussed this initially when this collection was made, and I personally don't like hard dependencies between roles. It makes things less flexible and does not allow one to opt-out of using a role.
Since we didn't agree on removing dependencies, I ended up fixing it in my local copy (after the dependencies messed some things up :D )

I do get the point of "adr is pointless without java". But one does not need jdk to run adr, and jre is enough if one is not doing java development, for example. This is just one of the examples where an opt-out would be nice and an explicit dependency makes things less flexible.

IMO for cases where a role has some requirements (i.e. `adr` command-line tool requires java), this should be mentioned in the role's documentation as a requirement, and not "fixed" via role dependency.

Reply:
I share the concern about users facing errors and having to support it, but I think the flexibility argument trumps that. It's like you say, the dependency really isn't on the java _role_, but just on _a_ java. For instance in platform, just installing the 'default-jre' would be more than enough.

One thing to note is that we generally have shared playbooks within teams that use these roles. Ensuring dependencies are done at that level. I don't think there are many people baking their own setup with these roles, and if they do, then they probably know what they are doing.

Reply:
Definitely agree with the above.
Most of our developers will use [devxp-linux](https://github.com/elhub/devxp-linux), which ensures all the needed roles are executed and "dependencies" are in place. And those who don't use the defaults, well they should be knowing what they're doing. And don't we have an "unwritten policy" on that, which goes somewhere along the lines of "if you do something differently - you're on your own" :D

Reply:
Maybe this PR should also include updates to documentation of the roles, to mention requirements for roles that have implicit dependency on something "external" to be present.

Reply:
Hmm. Not sure I agree 100%, but I acknowledge the argument. In this case, however, there should be documentation. That documentation should be added along with this commit, in that case (if it isn't, it'll never get added).

We should have a README in the affected roles (or it should be in the main README).